qcodes/utils/ndim_analysis/dataset/doNd.py

```diff
@@ -0,0 +1,309 @@
+        do_plot: should png and pdf versions of the images be saved after the
+            run.
+
+    Returns:
+        The run_id of the DataSet created
+    """
+
+    meas = Measurement()
+    _register_parameters(meas, (param_set1, param_set2))
+    _register_parameters(meas, param_meas, setpoints=(param_set1, param_set2))
+    _set_write_period(meas, write_period)
+    _register_actions(meas, enter_actions, exit_actions)
+
+    param_set1.post_delay = delay1
+    param_set2.post_delay = delay2
+
+    with _catch_keyboard_interrupts() as interrupted, meas.run() as datasaver:
+        for set_point1 in np.linspace(start1, stop1, num_points1):
+                if set_before_sweep:
+                    param_set2.set(start2)
+
+                param_set1.set(set_point1)
+                for action in before_inner_actions:
+                    action()
+                for set_point2 in np.linspace(start2, stop2, num_points2):
+                    # skip first inner set point if `set_before_sweep`
+                    if set_point2 == start2 and set_before_sweep:
+                        pass
+                    else:
+                        param_set2.set(set_point2)
+
+                    datasaver.add_result((param_set1, set_point1),
+                                         (param_set2, set_point2),
+                                         *_process_params_meas(param_meas))
+                for action in after_inner_actions:
+                    action()
+                if flush_columns:
+                    datasaver.flush_data_to_database()
+
+    return _handle_plotting(datasaver, do_plot, interrupted())
+
+
+
+
+def _handle_plotting(
+        datasaver: DataSaver,
+        do_plot: bool = True,
+        interrupted: bool = False
+) -> AxesTupleList:
+    """
+    Save the plots created by datasaver as pdf and png
+
+    Args:
+        datasaver: a measurement datasaver that contains a dataset to be saved
+            as plot.
+            :param do_plot:
+
+    """
+    dataid = datasaver.run_id
+    if do_plot == True:
+        res = _create_plots(datasaver)
+    else:
+        res = dataid, None, None
+
+    if interrupted:
+        raise KeyboardInterrupt
+
+    return res
+
+
+def _create_plots(datasaver: DataSaver) -> AxesTupleList:
+    dataid = datasaver.run_id
+    plt.ioff()
+    start = time.time()
+    axes, cbs = plot_by_id(dataid)
+    stop = time.time()
+    print(f"plot by id took {stop - start}")
+    mainfolder = config.user.mainfolder
+    experiment_name = datasaver._dataset.exp_name
+    sample_name = datasaver._dataset.sample_name
+    storage_dir = os.path.join(mainfolder, experiment_name, sample_name)
+    os.makedirs(storage_dir, exist_ok=True)
+    png_dir = os.path.join(storage_dir, 'png')
+    pdf_dif = os.path.join(storage_dir, 'pdf')
+    os.makedirs(png_dir, exist_ok=True)
+    os.makedirs(pdf_dif, exist_ok=True)
+    save_pdf = True
+    save_png = True
```

Comment:
Is it possible to add these to the config file so one can choose to save either png, pdf, or both?